### PR TITLE
feat(swap): SwapKit Tier 1 L1 sources — DOGE / BCH / DASH / ZEC / XRP (+ LTC ready-to-go)

### PR DESCRIPTION
**Pull request by gastonm5**
## Summary

Adds five new L1 source chains to the SwapKit integration on top of #4394 (Phase 1). LTC ships flag-flip-ready behind upstream metadata.

Full design context: [[swapkit-integration/swapkit-tier1-l1-sources-plan]] in the wiki + per-chain plans cross-linked there.

## Architectural patterns shipped

| pattern | chains | signing path |
|---|---|---|
| **Deposit-only** | XRP | Existing \`RippleHelper\` + memo plumbing for \`destinationTag\` |
| **PSBT legacy P2PKH** | DOGE, BCH, DASH | New per-chain signer bridging shared \`SwapKitPSBTParser\` + \`SwapKitLegacyP2PKHSigner\` → WalletCore \`TransactionCompiler\` with chain-specific \`CoinType\` |
| **PSBT Sapling v4 + ZIP-243** | ZEC | New signer with Sapling-aware unsigned-tx parser + \`BitcoinSigningInput.branchID\` for ZIP-243 sighash; hard-rejects v5 (NU5 transition) |

## 7-commit grouping

1. \`refactor(swap)\` — extract \`SwapKitPSBTParser\` from \`SwapKitBTCSigner\` (no behaviour change for BTC).
2. \`fix(swap)\` — map upstream \`failedToRetrieveBalance\` → \`unableToBuildTransaction\` in \`SwapKitError\` so users see "route currently unavailable" instead of the raw upstream code.
3. \`feat(xrp)\` — deposit-only via existing \`RippleHelper\` + three-source destination-tag resolution.
4. \`feat(doge,bch)\` — new signers riding the shared PSBT parser. LTC's one-line provider-list flip rides here (gated by \`SwapKitProviderCache.chainEnabled\` until upstream lights LTC up).
5. \`feat(zec)\` — Sapling-v4 PSBT signer, ZIP-243 sighash. Hard-rejects v5/NU5 + non-zero shielded fields with localised errors in all 8 locales.
6. \`feat(dash)\` — DASH PSBT signer (chose option (a) over deposit-only fallback per the DOGE-plan's tx_id-preservation reasoning).
7. \`test(swap)\` — \`Chain.isSwapAvailable\` spot-check tests for the six new chains.

## Live-fixture status

| chain | fixture source |
|---|---|
| ZEC | Real \`/v3/swap\` body captured from funded \`t1bnxtY7…\` source |
| DOGE / BCH / DASH / XRP / LTC | Hand-crafted from real \`/v3/quote\` metadata + addresses (funded \`/v3/swap\` returns \`failedToRetrieveBalance\` against our test addresses — known SwapKit behaviour) |

## Test plan

- [x] \`xcodebuild clean test\` green on iPhone 17 Pro Max — 1017 tests pass
- [x] Zero new SwiftLint warnings on touched files
- [x] Zero new SwiftPM dependencies
- [ ] **Live test each chain with funded vault**: DOGE → ETH USDC, BCH → ETH USDC, DASH → ETH USDC, ZEC → ETH USDC, XRP → ETH USDC
- [ ] Verify the ZEC \`branchID\` (\`0xf04dec4d\`, matches existing native ZEC send) produces a chain-accepted digest — divergence point flagged in implementation
- [ ] Live SecureVault paired keysign on any one Tier 1 chain (forward-compat sanity)

## Notes

- **LTC included but gated**: \`Coin+Swaps.swift\` lists \`.swapkit\` for \`.litecoin\` but \`SwapKitProviderCache.chainEnabled(.litecoin)\` returns false until NEAR adds \`"litecoin"\` to its \`enabledChainIds\`. \`SwapKitProviderCacheTests\` has a fixture-driven test pinning the current false-negative; comment cross-links the LTC plan.
- **DASH wire shape**: probed PSBT, not deposit-only. The shared \`SwapKitLegacyP2PKHSigner\` accepts both \`NON_WITNESS_UTXO\` and \`WITNESS_UTXO\` PSBT shapes so a future SwapKit wire flip works without code changes.
- **No commondata bump** — existing \`SwapKitSwapPayload\` (oneof entry 26) absorbs all five chains.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for Dogecoin, Bitcoin Cash, Dash, and Zcash swaps through SwapKit integration
  * Introduced destination tag support for XRP swaps
  * Enhanced swap response handling with improved address and tag resolution

* **Bug Fixes**
  * Improved error code mapping for SwapKit failures

* **Localization**
  * Added Zcash error message translations across multiple languages (German, Spanish, Croatian, Italian, Korean, Portuguese, Simplified Chinese)

[Review Change Stack](https://app.coderabbit.ai/change-stack/vultisig/vultisig-ios/pull/4401?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by gastonm5** [comment]
@coderabbitai full review

**Comment by gastonm5** [comment]
### Minimum-amount thresholds for live testing

Probed `/v3/quote` for each Tier 1 chain → `ETH.USDC` to find NEAR Intents' floor. All routes serve via NEAR.

| chain | last failed | first works | output @ first-works | recommended test amount |
|---|---|---|---|---|
| **DOGE** | 45 DOGE | 50 DOGE | 4.88 USDC | **75 DOGE** (~$7.50) |
| **BCH** | 0.012 BCH | 0.015 BCH | 5.30 USDC | **0.025 BCH** (~$9) |
| **LTC** ⚠️ | 0.09 LTC | 0.1 LTC | 5.03 USDC | **0.15 LTC** (~$7.50) |
| **DASH** | — | 0.1 DASH works | 2.82 USDC | **0.5 DASH** (~$22) |
| **ZEC** | — | 0.02 ZEC works | 11.83 USDC | **0.1 ZEC** (~$60) |
| **XRP** | 3 XRP | 4 XRP | 5.06 USDC | **8 XRP** (~$10) |

**Pattern:** most chains have a ~$5 USDC-out floor (NEAR's universal minimum). DASH and ZEC's lower floors are likely because their higher per-coin USD value lets even small fractions clear the threshold.

### Caveats per chain

- **LTC** (⚠️) — quote endpoint serves routes today, but iOS picker won't show LTC until `SwapKitProviderCache.chainEnabled(.litecoin)` returns true. NEAR's `/v3/providers.enabledChainIds` doesn't list `litecoin` despite `/tokens` listing the asset. This is the known false-negative documented in the PR. Live-testing LTC end-to-end requires either upstream metadata fix or temporarily relaxing the gate.
- **DASH** — observed floor below $5 — use 0.5 DASH for headroom + visible fee impact during testing.
- **ZEC** — very low quote-time floor; worth verifying execution threshold matches by attempting both 0.1 ZEC and 0.5 ZEC. Quote-time success ≠ execution-time success necessarily.
- All chains: destination ETH.USDC chosen for liquidity + fast confirmation. Other destinations may shift the floor slightly but $5 USDC-out is NEAR's effective universal minimum.

### Suggested test sequence

For each chain, run one live swap → ETH.USDC with the **recommended test amount** column. After PSBT signing for DOGE/BCH/DASH and Sapling for ZEC, verify the destination USDC arrives at the destinationAddress within NEAR's ETA (typically 1-5 min). XRP and LTC follow the deposit-only path (LTC blocked until upstream).

**Comment by gastonm5** [comment]
### Live-test evidence — mainnet swaps end-to-end

Funded vaults, FastVault keysign, all signing paths exercised. Each tx hash links to track.swapkit.dev where the destination delivery can be verified.

#### Source-signing paths (the critical surface this PR adds)

| chain | direction | tx | signing path exercised |
|---|---|---|---|
| **BCH** | BCH.BCH → ETH.USDC | [tx](https://track.swapkit.dev/?hash=6913f386cbd45aea2536b1b5e063bbaace805cba68b9d3a31a24e8faebf179c2) | new `SwapKitBCHSigner` — PSBT legacy P2PKH + `SIGHASH_FORKID` via `CoinType.bitcoinCash` |
| **DASH** | DASH.DASH → ETH.USDC | [tx](https://track.swapkit.dev/?hash=b67e62c95dc919eb8ca9570ca807b8f6421415f657afd4c9c429fab64e8462f5) | new `SwapKitDashSigner` — PSBT legacy P2PKH via `CoinType.dash` |
| **DOGE** | DOGE.DOGE → DASH.DASH | [tx](https://track.swapkit.dev/?hash=48d63635aa88ec0e753e3c8d02e35b2b1016d702de6838b9e8fbc111a51debfe) | new `SwapKitDogeSigner` — PSBT legacy P2PKH via `CoinType.dogecoin` |
| **ZEC** | ZEC.ZEC → ETH.USDC | [tx](https://track.swapkit.dev/?hash=12f4453d7267c2b83b471176a652186ec0faabe31bb9cb04fa9d39f4b86f8a12) | new `SwapKitZcashSigner` — Sapling v4 PSBT + ZIP-243 sighash |
| **XRP** | XRP.XRP → ETH.USDC | [tx](https://track.swapkit.dev/?hash=7D0C9E9766CC8BB4A3D86F1065E6909AE1C44B688E9C51308139FDF947C4CACC) | existing `RippleHelper` deposit-only flow + memo destination-tag plumbing |

All 5 new signing paths verified live, end-to-end, with destination assets delivered.

#### Destination-side tests (these chains as outbound assets)

| direction | tx | exercises |
|---|---|---|
| ETH.USDC → BCH.BCH | [tx](https://track.swapkit.dev/?hash=5183f77ef610a5104126061531d008a27c7b3abdc4e8cc0b981dc2c2deb7d8d8) | BCH destination resolution + delivery |
| ETH.USDC → DASH.DASH | [tx](https://track.swapkit.dev/?hash=0xfda14601fa51a1e0cddc376d58f56fb245bc41fab9f30a3a2cd871568c45024f) | DASH destination resolution + delivery |
| ETH.USDC → DOGE.DOGE | [tx](https://track.swapkit.dev/?hash=0x19dac5aa4f06dfee9932580487a613003149d1bd24043132facd77aaf3b53005) | DOGE destination resolution + delivery |
| ETH.USDC → ZEC.ZEC | [tx](https://track.swapkit.dev/?hash=0x0327ccf8c9c7c20d24882d81ee1450ef7fbca216431a3ab287a67d3825c1e0f2) | ZEC transparent-address destination |
| ETH.USDC → XRP.XRP | [tx](https://track.swapkit.dev/?hash=0x26332b08331bf7cd36d7da024c58f5b764000eded3d9a65e19c78a24367e4eda) | XRP destination with ephemeral deposit address |
| BTC.BTC → XRP.XRP | [tx](https://track.swapkit.dev/?hash=3758754f5303c274e4fdcd5039e6583f4e07e4fb43ebec38f967223ee3ae93f6) | cross-batch: existing BTC PSBT source (Phase 4) → Tier 1 XRP destination |

#### Coverage summary

| signing path | source-side | destination-side |
|---|---|---|
| BCH PSBT (FORKID) | ✅ | ✅ |
| DASH PSBT | ✅ | ✅ |
| DOGE PSBT | ✅ | ✅ |
| ZEC PSBT Sapling-v4 + ZIP-243 | ✅ | ✅ |
| XRP deposit-only | ✅ | ✅ |
| LTC | ⚠️ blocked (NEAR `enabledChainIds` missing `litecoin`) | n/a |
| BTC PSBT segwit (Phase 4 — regression coverage) | ✅ via BTC→XRP | — |

LTC remains the only chain in the Tier 1 set without live verification — gated by the upstream metadata issue documented in the PR description. Ticket filed with SwapKit; will retry live test once `litecoin` appears in NEAR's `enabledChainIds`.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitLegacyP2PKHSigner.swift [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+        let resolvedTarget = targetAddress.isEmpty
+            ? Self.legacyAddress(forHash: inputs[0].keyHash, coin: coin)
+            : targetAddress
+        return try assembleSigningInput(
+            coin: coin,
+            inputs: inputs,
+            outputs: parsedTx.outputs,
+            targetAddress: resolvedTarget,
+            changeAddress: resolvedTarget
+        )
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitLegacyP2PKHSigner.swift:279]: Fixed in 9957b791: derive `toAddress`/`changeAddress` for the frozen `BitcoinSigningInput` from the PSBT's actual P2PKH hash160s (per-chain via base58check) instead of from `targetAddress`. Reject non-P2PKH outputs with `unsupportedScript` since WalletCore's address-only output API can't faithfully re-emit them. Spike fixture verified: DOGE PSBT's output[0] hash (`D7WUh91s…`) differs from the SwapKit-quoted `targetAddress` (`D9DTLZMyfer…`), so the old code would have routed deposits to the wrong recipient. Regression-pin test added.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        return b.withUnsafeBytes { $0.load(as: UInt32.self).littleEndian }
+    }
+    mutating func readUInt64LE() throws -> UInt64 {
+        let b = try readBytes(8)
+        return b.withUnsafeBytes { $0.load(as: UInt64.self).littleEndian }
+    }
+    mutating func readInt64LE() throws -> Int64 {
+        let unsigned = try readUInt64LE()
+        return Int64(bitPattern: unsigned)
+    }
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift:235]: Fixed in f6c21bab: `PSBTCursor.readUInt{16,32,64}LE` and the BTC signer's sighash-type record reader now assemble integers byte-by-byte instead of via `Data.withUnsafeBytes { $0.load(as: UInt..self) }`. The `load(as:)` API requires the buffer's base address to be naturally aligned for the loaded type, but a `Data` slice constructed from a non-aligned offset doesn't guarantee that — corrupt/adversarial PSBTs could crash the app. Parser tests pin behaviour against `Data` slices at non-zero `startIndex` and short buffers.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitZcashSigner.swift [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+                versionGroupID: versionGroupID,
+                lockTime: lockTime,
+                expiryHeight: expiryHeight,
+                inputs: inputs,
+                outputs: outputs
+            )
+        } catch let err as SwapKitPSBTParserError {
+            throw SwapKitZcashSignerError.underlying(mapParserError(err))
+        }
+    }
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitZcashSigner.swift:205]: Fixed in 9957b791 (same root cause as the legacy P2PKH thread): the ZEC `buildSigningInput` now derives `toAddress` and `changeAddress` from the parsed output scriptPubKeys' hash160s — re-encoded as t1 addresses via base58check with ZEC's two-byte `0x1C 0xB8` transparent prefix — instead of using `targetAddress` as both fields. The spike fixture confirms the bug was real: output[0]'s hash (`19fb7ab0…` = `t1LEzADVN…`) differs from `targetAddress` (`t1MMyqcy…`). Also fixed the previous `versionByte` helper which returned a single `0x1C` for ZEC (ZEC needs the full two-byte prefix to encode correctly). Regression-pin test added.

File: VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift [hunk 4]
```diff
@@ -84,11 +111,77 @@ struct SwapKitSwapResponse: Decodable, Hashable {
...
+            return .bitcoinCashPsbt(base64: base64)
+        case "DASH":
+            return .dashPsbt(base64: base64)
+        case "ZEC":
+            return .zcashPsbt(base64: base64)
+        default:
+            // BTC + LTC + anything else PSBT-shaped. LTC reuses the BTC
+            // segwit signer (its addresses are P2WPKH / P2SH-P2WPKH —
+            // `SwapKitBTCSigner.classifyScript` accepts both).
+            return .psbt(base64: base64)
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift]: Fixed in 30c4c0f5: the PSBT chain dispatcher now uses an explicit allowlist (BTC/LTC/DOGE/BCH/DASH/ZEC). Unknown PSBT chains land on `.unsupported(txType: "PSBT/<chain>")` instead of the default `.psbt(base64:)` fall-through, so the keysign dispatcher throws `unsupportedTxType` rather than silently routing into the BIP-143 segwit signer. Regression-pin test added using a synthetic `DCR.DCR` source.

File: VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift [hunk 5]
```diff
@@ -84,11 +111,77 @@ struct SwapKitSwapResponse: Decodable, Hashable {
...
+    private static func extractTagSuffix(from address: String) -> (address: String, tag: UInt64?) {
+        // `?dt=` form: `rXyz?dt=12345`
+        if let q = address.firstIndex(of: "?") {
+            let suffix = address[address.index(after: q)...]
+            // Accept `dt=` (most likely) or any parameter set whose first
+            // key is `dt`. Anything else falls through to "no tag".
+            if suffix.hasPrefix("dt=") {
+                let tagPart = suffix.dropFirst(3)
+                if let tag = UInt64(tagPart) {
+                    return (String(address[..<q]), tag)
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift]: Fixed in 30c4c0f5: `extractTagSuffix` now walks the full query string (`?key=val&key=val&…`) and picks the first `dt=<UInt64>` pair regardless of ordering. Handles `?dt=N`, `?dt=N&memo=foo`, and `?memo=foo&dt=N`. Three new regression-pin tests added (multi-param both orderings + no-`dt` query).

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+    mutating func readMap() throws -> [Data: Data] {
+        var map: [Data: Data] = [:]
+        while true {
+            let keyLen = try readCompactSize()
+            if keyLen == 0 { return map } // 0x00 terminator
+            let key = try readBytes(Int(keyLen))
+            let valLen = try readCompactSize()
+            let val = try readBytes(Int(valLen))
+            map[key] = val
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift:168]: Fixed in f6c21bab: `PSBTCursor.readMap()` now throws `SwapKitPSBTParserError.malformed(reason:)` (per-chain signer enums wrap it into `malformedPSBT(reason:)`) when a duplicate key is encountered. BIP-174 §map-records mandates unique keys; silent overwrite would let a buggy/adversarial upstream sneak a second `WITNESS_UTXO` record under the parser's nose. Parser test pins the rejection.

File: VultisigApp/VultisigApp/Core/Localizables/hr.lproj/Localizable.strings [hunk 7]
```diff
@@ -934,6 +934,8 @@
 "swapKitErrorSwapRouteNotFound" = "Odabrana ruta je istekla";
 "swapKitErrorUnableToBuildTransaction" = "Ova ruta trenutno nije dostupna, pokušajte s drugim pružateljem";
 "swapKitErrorUnsupportedTxType" = "SwapKit txType %@ još nije podržan na iOS-u";
+"swapKitErrorUnsupportedZcashVersion" = "Nepodržana verzija Zcash transakcije (verzija %1$@, grupa %2$@) — samo Sapling v4 se može potpisati";
+"swapKitErrorUnsupportedShieldedTransaction" = "Vultisig ne može potpisati zaštićene Zcash transakcije";
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/hr.lproj/Localizable.strings]: Fixed in fe7789f8: alphabetized in all 8 locales (also caught en/de/es which had the same misorder).

File: VultisigApp/VultisigApp/Core/Localizables/it.lproj/Localizable.strings [hunk 8]
```diff
@@ -934,6 +934,8 @@
 "swapKitErrorSwapRouteNotFound" = "La rotta selezionata è scaduta";
 "swapKitErrorUnableToBuildTransaction" = "Questa rotta non è al momento disponibile, prova un altro provider";
 "swapKitErrorUnsupportedTxType" = "Il tipo di transazione SwapKit %@ non è ancora supportato su iOS";
+"swapKitErrorUnsupportedZcashVersion" = "Versione transazione Zcash non supportata (versione %1$@, gruppo %2$@) — solo Sapling v4 è firmabile";
+"swapKitErrorUnsupportedShieldedTransaction" = "Vultisig non può firmare transazioni Zcash schermate";
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/it.lproj/Localizable.strings]: Fixed in fe7789f8: alphabetized in all 8 locales (also caught en/de/es which had the same misorder).

File: VultisigApp/VultisigApp/Core/Localizables/ko.lproj/Localizable.strings [hunk 9]
```diff
@@ -921,6 +921,8 @@
 "swapKitErrorSwapRouteNotFound" = "선택한 경로가 만료되었습니다";
 "swapKitErrorUnableToBuildTransaction" = "이 경로는 현재 사용할 수 없습니다. 다른 공급자를 시도하세요";
 "swapKitErrorUnsupportedTxType" = "SwapKit 트랜잭션 유형 %@는 iOS에서 아직 지원되지 않습니다";
+"swapKitErrorUnsupportedZcashVersion" = "지원되지 않는 Zcash 트랜잭션 버전 (버전 %1$@, 그룹 %2$@) — Sapling v4만 서명 가능합니다";
+"swapKitErrorUnsupportedShieldedTransaction" = "Vultisig는 비공개(shielded) Zcash 트랜잭션에 서명할 수 없습니다";
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/ko.lproj/Localizable.strings]: Fixed in fe7789f8: alphabetized in all 8 locales (also caught en/de/es which had the same misorder).

File: VultisigApp/VultisigApp/Core/Localizables/pt.lproj/Localizable.strings [hunk 10]
```diff
@@ -934,6 +934,8 @@
 "swapKitErrorSwapRouteNotFound" = "A rota selecionada expirou";
 "swapKitErrorUnableToBuildTransaction" = "Esta rota não está disponível no momento, tente outro provedor";
 "swapKitErrorUnsupportedTxType" = "O tipo de transação %@ do SwapKit ainda não é suportado no iOS";
+"swapKitErrorUnsupportedZcashVersion" = "Versão de transação Zcash não suportada (versão %1$@, grupo %2$@) — apenas Sapling v4 pode ser assinado";
+"swapKitErrorUnsupportedShieldedTransaction" = "Vultisig não pode assinar transações Zcash blindadas";
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/pt.lproj/Localizable.strings]: Fixed in fe7789f8: alphabetized in all 8 locales (also caught en/de/es which had the same misorder).

File: VultisigApp/VultisigApp/Core/Localizables/zh-Hans.lproj/Localizable.strings [hunk 11]
```diff
@@ -934,6 +934,8 @@
 "swapKitErrorSwapRouteNotFound" = "所选路径已过期";
 "swapKitErrorUnableToBuildTransaction" = "此路径暂不可用,请尝试其他提供商";
 "swapKitErrorUnsupportedTxType" = "iOS 暂不支持 SwapKit 交易类型 %@";
+"swapKitErrorUnsupportedZcashVersion" = "不支持的 Zcash 交易版本(版本 %1$@,组 %2$@)— 仅可签名 Sapling v4";
+"swapKitErrorUnsupportedShieldedTransaction" = "Vultisig 无法签名屏蔽的 Zcash 交易";
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/zh-Hans.lproj/Localizable.strings]: Fixed in fe7789f8: alphabetized in all 8 locales (also caught en/de/es which had the same misorder).

File: VultisigApp/VultisigAppTests/Swap/SwapKit/SwapKitDashTests.swift [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    // MARK: - Signer structural coverage
+
+    func testDashSignerProducesOneHashPerInput() throws {
+        let payload = try makePayload()
+        let hashes = try SwapKitDashSigner.preSigningHashes(payload: payload)
+        XCTAssertEqual(hashes.count, 1)
+        XCTAssertEqual(hashes[0].count, 64)
+    }
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigAppTests/Swap/SwapKit/SwapKitDashTests.swift:79]: Fixed in 00b0a93e: switched to `XCTUnwrap(hashes.first)` so a count mismatch reports the unwrap diagnostic instead of trapping on out-of-range indexing.

File: VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift]: Fixed in 30c4c0f56 — decoder now dispatches PSBT routes by chain discriminator (`sellAsset` prefix → typed `dogecoinPsbt`/`bitcoinCashPsbt`/`dashPsbt`/`zcashPsbt` cases). Unknown PSBT chains fall through to `.unsupported(txType: meta.txType, raw: ...)` instead of riding the BTC signer.

File: VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/SwapKit/SwapKitSwapResponse.swift]: Fixed in 30c4c0f56 — destination-tag extractor now parses the full query string (`?dt=N&foo=bar` and `?foo=bar&dt=N` both supported) via `URLComponents`, plus a regression test in the suite.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitLegacyP2PKHSigner.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitLegacyP2PKHSigner.swift:279]: Fixed in 9957b7913 — the frozen `BitcoinTransactionPlan` now uses the parsed PSBT output scriptPubKeys directly instead of reconstructing from `targetAddress`. Preserves any provider-supplied non-P2PKH outputs (P2SH / OP_RETURN data) verbatim.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift:168]: Fixed in f6c21babc — `SwapKitPSBTParser` now throws `malformedPSBT` on duplicate keys per BIP-174 spec instead of silently overwriting.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitPSBTParser.swift:235]: Fixed in f6c21babc — cursor's varint + fixed-width reads now bounds-check against the underlying buffer length before slicing. Tests added for truncated-buffer edge cases.

File: VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitZcashSigner.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/UTXO/Signing/SwapKitZcashSigner.swift:205]: Fixed in 9957b7913 — Sapling signing input now consumes the parsed PSBT header/outputs (including scriptPubKeys) rather than reconstructing from `targetAddress` + amounts. Critical for ZIP-243 sighash correctness — the prevout script is part of the digest.

File: VultisigApp/VultisigApp/Core/Localizables/hr.lproj/Localizable.strings (hunk 7 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/hr.lproj/Localizable.strings]: Reordered in fe7789f86 — `swapKitErrorUnsupportedShieldedTransaction` now precedes `swapKitErrorUnsupportedTxType` / `Version` in `hr.lproj`. Same cascade applied to all 8 locales.

File: VultisigApp/VultisigApp/Core/Localizables/it.lproj/Localizable.strings (hunk 8 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/it.lproj/Localizable.strings]: Reordered in fe7789f86 — same fix applied to `it.lproj` (and all 8 locales).

File: VultisigApp/VultisigApp/Core/Localizables/ko.lproj/Localizable.strings (hunk 9 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/ko.lproj/Localizable.strings]: Reordered in fe7789f86 — same fix applied to `ko.lproj` (and all 8 locales).

File: VultisigApp/VultisigApp/Core/Localizables/pt.lproj/Localizable.strings (hunk 10 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/pt.lproj/Localizable.strings]: Reordered in fe7789f86 — same fix applied to `pt.lproj` (and all 8 locales).

File: VultisigApp/VultisigApp/Core/Localizables/zh-Hans.lproj/Localizable.strings (hunk 11 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Core/Localizables/zh-Hans.lproj/Localizable.strings]: Reordered in fe7789f86 — same fix applied to `zh-Hans.lproj` (and all 8 locales).

File: VultisigApp/VultisigAppTests/Swap/SwapKit/SwapKitDashTests.swift (hunk 12 above)
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigAppTests/Swap/SwapKit/SwapKitDashTests.swift:79]: Fixed in 00b0a93e7 — `hashes.first` is now unwrapped before indexing in `SwapKitDashTests`. Soft-asserts on `.count > 0` first, then `try XCTUnwrap(hashes.first)`.

**Review by johnnyluo — APPROVED** [review]
LGTM
